pkg/worker/log/logger.go

```diff
@@ -0,0 +1,131 @@
+package log
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+)
+
+// Logger implementation.
+type Logger struct {
+	config  conf.Config
+	watcher Watcher
+}
+
+// NewLogger returns a new Logger instance.
+func NewLogger(config conf.Config) Logger {
+	return Logger{
+		config:  config,
+		watcher: NewWatcher(),
+	}
+}
+
+// Path returns a absolute file path given a name.
+func (l *Logger) Path(name string) string {
+	return filepath.Join(l.config.LogFolder, fmt.Sprintf("%s.log", name))
+}
+
+// Create creates and return a os.File under log folder.
+// If the file can't be created an error will be returned.
+func (l *Logger) Create(name string) (*os.File, error) {
+	return os.Create(l.Path(name))
+}
+
+// Remove removes the named file under the log folder.
+func (l *Logger) Remove(name string) error {
+	return os.Remove(l.Path(name))
+}
+
+// Tailf watch a named log file under the log folder, and
+// streams his content through a channel.
+func (l *Logger) Tailf(ctx context.Context, name string) (chan string, error) {
+	file, err := os.OpenFile(l.Path(name), os.O_RDONLY, 0644)
+	if err != nil || os.IsNotExist(err) {
+		return nil, err
+	}
+	logchan := make(chan string)
+	go func() {
+		defer func() {
+			file.Close()
+			close(logchan)
+		}()
+		// reads file from the begin
+		offset := int64(0)
```

Comment:
Hmm, I may be missing something, but it seems like `offset` is not really used anymore? You're not doing Seek anymore.

Reply:
I'm using the offset in streamFile() function, as a parameter. I can eliminate this if I move bufio.NewReader to outside the streamFile(). In this way, we don't need to do Seek in the file anymore.